src/main/java/org/folio/persist/HoldingsRepository.java

```diff
@@ -19,6 +21,53 @@ public HoldingsRepository(Context context, Map<String, String> okapiHeaders) {
     super(postgresClient(context, okapiHeaders), HOLDINGS_RECORD_TABLE, HoldingsRecord.class);
   }
 
+  /**
+   * Row where the first value is an array with all holdings records; the second value is
+   * the exact totalRecords count.
+   */
+  public Future<Row> getByInstanceId(String instanceId, String[] sortBys, int offset, int limit) {
+    var orderBy = new StringBuilder();
+    for (var sortBy : sortBys) {
+      if (sortBy.isEmpty()) {
+        continue;
+      }
+      var s = switch (sortBy) {
+        case "effectiveLocation.name" -> "name";
+        case "callNumberPrefix",
+             "callNumber",
+             "callNumberSuffix" -> "jsonb->>'" + sortBy + "'";
+        default -> null;
+      };
+      if (s == null) {
+        return Future.failedFuture(new IllegalArgumentException("sortBy: " + sortBy));
+      }
+      if (!orderBy.isEmpty()) {
+        orderBy.append(", ");
+      }
+      orderBy.append(s);
+    }
+    var sql = "WITH data AS ("
+        + " SELECT h.jsonb AS jsonb, l.jsonb->>'name' AS name"
+        + " FROM " + HOLDINGS_RECORD_TABLE + " h"
+        + " LEFT JOIN " + LOCATION_TABLE + " l"
+        + "   ON h.effectiveLocationId = l.id"
+        + " WHERE h.instanceId=$1"
+        + " )"
```

Comment:
Personal opinion: if this CTE is dealing with enough data to justify the `OFFSET` and `LIMIT`, then it should probably be set to not get materialized.

```suggestion
    var sql = "WITH data AS NOT MATERIALIZED ("
        + " SELECT h.jsonb AS jsonb, l.jsonb->>'name' AS name"
        + " FROM " + HOLDINGS_RECORD_TABLE + " h"
        + " LEFT JOIN " + LOCATION_TABLE + " l"
        + "   ON h.effectiveLocationId = l.id"
        + " WHERE h.instanceId=$1"
        + " )"
```